web_poet/mixins.py

```diff
@@ -78,3 +81,30 @@ def urljoin(self, url: str) -> str:
         url and baseurl of the response"""
         # FIXME: move it to HttpResponse
         return urljoin(self.base_url, url)
+
+
+class _UrlMixin(abc.ABC):
+    """Common shortcut methods for working with the URL of HTTP requests and
+    responses.
+
+    This mixin could be used with Page Object base classes.
+
+    It requires the ``url`` attribute to be present.
+    """
+
+    _cached_base_url = None
```

Comment:
For mixin it could make sense to use `__cached_base_url` name, so that there won't be a name clash in a subclass